src/main/java/org/cactoos/io/Zip.java

```diff
@@ -33,50 +33,58 @@
 import java.util.zip.ZipEntry;
 import java.util.zip.ZipOutputStream;
 import org.cactoos.Input;
+import org.cactoos.func.ForEach;
+import org.cactoos.scalar.ItemAt;
+import org.cactoos.text.Joined;
 
 /**
  * Zip files and directory.
  * <br>
  * <br>There is no thread-safety guarantee.
  *
  * @since 0.29
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
  */
 @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
 public final class Zip implements Input {
 
     /**
-     * Origin directory.
+     * Origin list of files.
      */
-    private final Directory origin;
+    private final Directory directory;
 
     /**
      * Ctor.
      *
-     * @param origin Origin directory.
+     * @param dir Origin directory.
      */
-    public Zip(final Directory origin) {
-        this.origin = origin;
+    public Zip(final Directory dir) {
+        this.directory = dir;
     }
 
     @Override
     public InputStream stream() throws Exception {
         final ByteArrayOutputStream out = new ByteArrayOutputStream();
+        final Path parent = new ItemAt<>(0, this.directory).value().getParent();
```

Comment:
@fanifieiev what if it is empty? Please add a test to validate this will work. I believe there is some need to change the interface `Directory` to expose the `Path` itself...

Reply:
@victornoel It is never empty. If you pass `Directory `object, it has a path and it has a parent folder:

- For linux: the last parent would be `/`
- For win: the last parent would be `C` or `D`, etc
 Regarding the `I believe there is some need to change the interface Directory to expose the Path itself...`, yes it is required and I will leave a todo.

Reply:
@fanifieiev can you add a test where the directory is empty at least?

Reply:
@victornoel Will do.